### PR TITLE
Feature/shared robot feedback type

### PR DESCRIPTION
**Pull request by Florisheinen1**
This PR makes use of the new Robot Feedback data type.
This PR is dependent on:
- [PR in utils](https://github.com/RoboTeamTwente/roboteam_utils/pull/111)
- [PR in Networking](https://github.com/RoboTeamTwente/roboteam_networking/pull/9)
- [PR in AI](https://github.com/RoboTeamTwente/roboteam_ai/pull/1340)
- [PR in World](https://github.com/RoboTeamTwente/roboteam_world/pull/78)

### CODE REVIEW

File: src/RobotHub.cpp [hunk 1]
```diff
@@ -227,49 +227,63 @@ void RobotHub::onSimulationConfiguration(const proto::SimulationConfiguration &c
...
-    feedbackToBePublished.set_isyellow(feedback.color == utils::TeamColor::YELLOW);
-
-    // proto::RobotFeedback* feedbackOfRobots = feedbackToBePublished.mutable_receivedfeedback();
-
-    for (auto const &[robotId, hasBall] : feedback.robotIdHasBall) {
-        proto::RobotFeedback *feedbackOfRobot = feedbackToBePublished.add_receivedfeedback();
-        feedbackOfRobot->set_id(robotId);
-        feedbackOfRobot->set_hasball(hasBall);
+    rtt::RobotsFeedback robotsFeedback;
+    robotsFeedback.source = rtt::RobotFeedbackSource::BASESTATION;
```
Comment (alexander-de-ranitz) [inline src/RobotHub.cpp]: This should be rtt::RobotFeedbackSource::SIMULATOR right?

File: src/RobotHub.cpp [hunk 2]
```diff
@@ -227,49 +227,63 @@ void RobotHub::onSimulationConfiguration(const proto::SimulationConfiguration &c
...
+    robotsFeedback.team = feedback.color == utils::TeamColor::YELLOW ? rtt::Team::YELLOW : rtt::Team::BLUE;
```
Comment (alexander-de-ranitz) [inline src/RobotHub.cpp]: What is the difference between utils::TeamColor and rtt::Team? Do we really need both?

File: src/RobotHub.cpp (hunk 1 above)
Comment (Florisheinen1) [reply src/RobotHub.cpp]: Yes, my bad, fixed it

File: src/RobotHub.cpp (hunk 2 above)
Comment (Florisheinen1) [reply src/RobotHub.cpp]: Previously, both AI and RobotHub had to define their own Team enum. I added a Teams enum in utils that can be used in all repos, but I did not remove the one in RobotHub yet. Ill do that now

File: src/RobotHub.cpp [hunk 3]
```diff
@@ -219,57 +218,71 @@ void RobotHub::onSimulationConfiguration(const proto::SimulationConfiguration &c
...
-
-    // proto::RobotFeedback* feedbackOfRobots = feedbackToBePublished.mutable_receivedfeedback();
-
-    for (auto const &[robotId, hasBall] : feedback.robotIdHasBall) {
-        proto::RobotFeedback *feedbackOfRobot = feedbackToBePublished.add_receivedfeedback();
-        feedbackOfRobot->set_id(robotId);
-        feedbackOfRobot->set_hasball(hasBall);
+    rtt::RobotsFeedback robotsFeedback;
+    robotsFeedback.source = rtt::RobotFeedbackSource::SIMULATOR;
+    robotsFeedback.team = feedback.color == rtt::Team::YELLOW ? rtt::Team::YELLOW : rtt::Team::BLUE;
```
Comment (alexander-de-ranitz) [inline src/RobotHub.cpp]: Can't you just say robotsFeedback.team = feedback.color?

File: src/RobotHub.cpp [hunk 4]
```diff
@@ -219,57 +218,71 @@ void RobotHub::onSimulationConfiguration(const proto::SimulationConfiguration &c
...
-    feedbackOfRobot->set_yaw(feedback.angle);
-    // feedbackOfRobot->set_batterylevel(feedback.batteryLevel); // Not in proto yet
-    feedbackOfRobot->set_signalstrength(feedback.rssi);
-    feedbackOfRobot->set_haslockedwheel(feedback.wheelLocked);
-
-    this->sendRobotFeedback(feedbackToBePublished);
+void RobotHub::handleRobotFeedbackFromBasestation(const REM_RobotFeedback &feedback, rtt::Team basestationColor) {
+    rtt::RobotsFeedback robotsFeedback;
+    robotsFeedback.source = rtt::RobotFeedbackSource::BASESTATION;
+    robotsFeedback.team = basestationColor == rtt::Team::YELLOW ? rtt::Team::YELLOW : rtt::Team::BLUE;
```
Comment (alexander-de-ranitz) [inline src/RobotHub.cpp]: Same as above: is this check necessary?

**Review by alexander-de-ranitz — APPROVED** [review]
lgtm

**Pull request merged**